src/zmalloc.c

```diff
@@ -504,12 +506,16 @@ void zfree_dram(void *ptr) {
 
     if (ptr == NULL) return;
 #ifdef HAVE_MALLOC_SIZE
+#ifndef USE_MEMKIND_MEMTIER
     update_zmalloc_dram_stat_free(zmalloc_size(ptr));
+#endif
     free_dram(ptr);
 #else
     realptr = (char*)ptr-PREFIX_SIZE;
     oldsize = *((size_t*)realptr);
```

Comment:
You don't use oldsize variable in case of USE_MEMKIND_MEMTIER consider this in ifdef block

Reply:
Done.